Automattic/jetpack
# Hide the plugin banner on non-wpcom-connected users or agency-managed users

Fixes https://github.com/Automattic/dotcom-forge/issues/8422

## Proposed changes:

This PR changes to hide the plugin banner on non-wpcom-connected users or agency-managed users.

| before | after  |
|--------|--------|
| <img width="1390" alt="Screenshot 2024-07-26 at 15 08 52" src="https://github.com/user-attachments/assets/04ac52d7-e7dc-4aba-8587-10443e6d82db"> | <img width="1394" alt="Screenshot 2024-07-26 at 15 07 50" src="https://github.com/user-attachments/assets/febc5967-a7e6-4f99-9bd1-4009367ef721"> | 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Prepare an Atomic Classic site
* Patch mu-wpcom change via https://github.com/Automattic/jetpack/pull/38532#issuecomment-2249533772
* Patch wpsomsh change by using `pnpm jetpack rsync wpcomsh YOURSITE@sftp.wp.com:htdocs/wp-content/mu-plugins/wpcoms`
	* Patching WordPress.com Site Helper by using Jetpack Beta errored out for me 
* Go to "Plugins > Add New Plugin"
	* Observe the banner is displayed
* Set `is_fully_managed_agency_site=1` option on the site
	* Observe the banner is not displayed
	* Revert `is_fully_managed_agency_site` to `0`
* Add a new user via "Users > Add New User" (don't connect it to a wpcom account)
	* Login with that added user
	* Go to "Plugins > Add New Plugin"
	* Observe the banner is not displayed
	* Connect the account with wpcom (e.g., via Jetpack > My Jerpack > "Some features require authentication. Sign in")
	* Go to "Plugins > Add New Plugin"
	* Observe the banner is displayed
